code/modules/reagents/Chemistry-Recipes.dm

```diff
@@ -1513,6 +1513,24 @@
 	var/obj/effect/golemrune/Z = new /obj/effect/golemrune
 	Z.forceMove(get_turf(holder.my_atom))
 
+/datum/chemical_reaction/soap_key
+	name = "Soap Key"
+	id = "skey"
+	result = null
+	required_reagents = list("frostoil" = 2, "cleaner" = 5)

```

Comment:
Perhaps frost oil is a bit overkill here considering how rare the chemical actually is.

Reply:
I suggest using triglycerides as one of the ingredients. Most soap is made with oil/fat.

One of the other ingredients is lye, which is from ashes. If you're willing to make a new chem to go along with this that other servers have, that would be going above and beyond but I think an alternative could possibly be salt or something.

Water is also common in soap as well.

I suggest: 1 part triglycerides, 1 part water, 1 part space cleaner. These are easy to obtain chemicals.